### PR TITLE
Documentation about loading a fast tokenizer within Transformers

### PR DESCRIPTION
**Pull request by LysandreJik**
This PR does two things:
- Allows to load a fast tokenizer from an instantiated `tokenizers` object
- Adds a page to document how to use these tokenizers within `transformers`

See [here](https://190138-155220641-gh.circle-artifacts.com/0/docs/_build/html/fast_tokenizers.html) for the generated docs

### CODE REVIEW

File: docs/source/fast_tokenizers.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
+Using tokenizers from 🤗 tokenizers
```
Comment (sgugger) [inline docs/source/fast_tokenizers.rst:1]:
```suggestion
Using tokenizers from 🤗 Tokenizers
```

File: docs/source/fast_tokenizers.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+=======================================================================================================================
+
+The :class:`~transformers.PreTrainedTokenizerFast` depends on the `tokenizers
+<https://huggingface.co/docs/tokenizers>`__ library. The tokenizers obtained from the 🤗 tokenizers library can be
+loaded very simply into 🤗 transformers.
```
Comment (sgugger) [inline docs/source/fast_tokenizers.rst:6]:
```suggestion
<https://huggingface.co/docs/tokenizers>`__ library. The tokenizers obtained from the 🤗 Tokenizers library can be
loaded very simply into 🤗 Transformers.
```

File: docs/source/fast_tokenizers.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    >>> files = [...]
+    >>> tokenizer.train(files, trainer)
+
+We now have a tokenizer trained on the files we defined. We can either continue using it in that runtime, or save it to
+a JSON file for future re-use.
+
+Loading directly from the tokenizer object
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Let's see how to leverage this tokenizer object in the 🤗 transformers library. The
```
Comment (sgugger) [inline docs/source/fast_tokenizers.rst:30]:
```suggestion
Let's see how to leverage this tokenizer object in the 🤗 Transformers library. The
```

File: docs/source/fast_tokenizers.rst [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+`tokenizer` object as an argument:
+
+.. code-block::
+
+    >>> from transformers import PreTrainedTokenizerFast
+
+    >>> fast_tokenizer = PreTrainedTokenizerFast(tokenizer_object=tokenizer)
+
+This object can now be used with all the methods shared by the 🤗 Transformers tokenizers! Head to `the tokenizer page
+<main_classes/tokenizer.html>`__ for more information.
```
Comment (sgugger) [inline docs/source/fast_tokenizers.rst:41]:
```suggestion
This object can now be used with all the methods shared by the 🤗 Transformers tokenizers! Head to :doc:`the tokenizer page
<main_classes/tokenizer>` for more information.
```

File: docs/source/fast_tokenizers.rst [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+method using the :obj:`tokenizer_file` parameter:
+
+.. code-block::
+
+    >>> from transformers import PreTrainedTokenizerFast
+
+    >>> fast_tokenizer = PreTrainedTokenizerFast(tokenizer_file="tokenizer.json")
+
+This object can now be used with all the methods shared by the 🤗 Transformers tokenizers! Head to `the tokenizer page
+<main_classes/tokenizer.html>`__ for more information.
```
Comment (sgugger) [inline docs/source/fast_tokenizers.rst:62]:
```suggestion
This object can now be used with all the methods shared by the 🤗 Transformers tokenizers! Head to :doc:`the tokenizer page
<main_classes/tokenizer>` for more information.
```

File: docs/source/main_classes/tokenizer.rst [hunk 6]
```diff
@@ -62,6 +62,11 @@ PreTrainedTokenizer
 PreTrainedTokenizerFast
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+The :class:`~transformers.PreTrainedTokenizerFast` depend on the `tokenizers
+<https://huggingface.co/docs/tokenizers>`__ library. The tokenizers obtained from the 🤗 tokenizers library can be
+loaded very simply into 🤗 transformers. Take a look at the `Using tokenizers from 🤗 tokenizers
+<../fast_tokenizers.html>`__ page to understand how this is done.
```
Comment (sgugger) [inline docs/source/main_classes/tokenizer.rst:68]:
```suggestion
loaded very simply into 🤗 transformers. Take a look at the :doc:`Using tokenizers from 🤗 tokenizers
<../fast_tokenizers>` page to understand how this is done.
```

File: src/transformers/tokenization_utils_fast.py [hunk 7]
```diff
@@ -54,6 +54,12 @@
 # Slow tokenizers have an additional added tokens files
 ADDED_TOKENS_FILE = "added_tokens.json"
 
+INIT_TOKENIZER_DOCSTRING += """
+        tokenizer_object (:class:`tokenizers.Tokenizer`):
+            A :class:`tokenizers.Tokenizer` object from 🤗 tokenizers to instantiate from. See `Using tokenizers from 🤗
+            tokenizers <../fast_tokenizers.html>`__ for more information.
```
Comment (sgugger) [inline src/transformers/tokenization_utils_fast.py:60]:
```suggestion
            A :class:`tokenizers.Tokenizer` object from 🤗 tokenizers to instantiate from. See :doc:`Using tokenizers from 🤗
            tokenizers <../fast_tokenizers>` for more information.
```

**Review by sgugger — APPROVED** [review]
Great addition! For the documentation, when linking to other pages of the doc it's best to use the `:doc:` syntax as it works across versions.

**Comment by LysandreJik** [comment]
Thanks for the suggestion @sgugger!

**Review by n1t0 — COMMENTED** [review]
:heart:

**Pull request merged**